src/annotations/components/annotationForm/AnnotationForm.tsx

```diff
@@ -0,0 +1,182 @@
+
+// Contexts
+import {OverlayContext} from 'src/overlays/components/OverlayController'
+
+interface Props {
+  title: 'Edit' | 'Add'
+  type: AnnotationType
+  timeStart: string
+  timeStop: string
+  summaryText?: string
+  messageText?: string
+  streamID?: string
+  onSubmit: (Annotation: Annotation) => void
+}
+
+const AnnotationForm: FC<Props> = ({
+  title,
+  type,
+  timeStart,
+  timeStop,
+  summaryText,
+  messageText,
+  streamID,
+  onSubmit,
+}) => {
+  const initialAnnotationState = getInitialAnnotationState(
+    type,
+    timeStart,
+    timeStop,
+    summaryText,
+    messageText,
+    streamID
+  )
+
+  const [state, dispatch] = useReducer(
+    annotationReducer,
+    initialAnnotationState
+  )
+
+  const {onClose} = useContext(OverlayContext)
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>): void => {
+    e.preventDefault()
+
+    const formIsValid = annotationFormIsValid(state, dispatch)
+
+    if (formIsValid) {
+      onSubmit(getAnnotationFromDraft(state))
+      onClose()
+    }
+  }
+
+  const handleSummaryChange = (e: ChangeEvent<InputRef>): void => {
+    dispatch({type: 'updateSummary', payload: e.target.value})
+  }
+
+  const handleTypeChange = (payload: AnnotationType): void => {
+    dispatch({type: 'updateType', payload})
+  }
+
+  const handleTimeStartChange = (e: ChangeEvent<InputRef>): void => {
+    dispatch({type: 'updateTimeStart', payload: e.target.value})
+  }
+
+  const handleTimeStopChange = (e: ChangeEvent<InputRef>): void => {
+    dispatch({type: 'updateTimeStop', payload: e.target.value})
+  }
+
+  const handleMessageChange = (e: ChangeEvent<TextAreaRef>): void => {
+    dispatch({type: 'updateMessage', payload: e.target.value})
+  }
+
+  const handleStreamChange = (payload: AnnotationStream): void => {
+    dispatch({type: 'updateStream', payload})
+  }
+
+  return (
+    <Overlay.Container maxWidth={560}>
+      <Overlay.Header title={`${title} Annotation`} onDismiss={onClose} />
+      <Form onSubmit={handleSubmit}>
+        <Overlay.Body>
+          <Grid>
+            <Grid.Row>
+              <Grid.Column widthXS={Columns.Seven}>
+                <AnnotationSummaryInput
+                  {...state.summary}
+                  onChange={handleSummaryChange}
+                />
+              </Grid.Column>
+              <Grid.Column widthXS={Columns.Five}>
+                <AnnotationTypeToggle
+                  type={state.type}
+                  onChange={handleTypeChange}
+                />
+              </Grid.Column>
+            </Grid.Row>
+            <Grid.Row>
+              <Grid.Column
+                widthXS={state.type === 'range' ? Columns.Six : Columns.Twelve}
+              >
+                <AnnotationTimeStartInput
+                  {...state.timeStart}
+                  type={state.type}
+                  onChange={handleTimeStartChange}
+                />
+              </Grid.Column>
+              {state.type === 'range' && (
```

Comment:
my inner watts is yelling about this not being included in a component that self guards

Reply:
and it is strange that it's not following the convention of the AnnotationTimeStartInput above?